app/services/socket.js

```diff
@@ -0,0 +1,43 @@
+import PhoenixSocket from 'phoenix/services/phoenix-socket';
+import { get } from '@ember/object';
+import { inject as service } from '@ember/service';
+import ENV from 'code-corps-ember/config/environment';
+
+export default PhoenixSocket.extend({
+  session: service(),
+
+  init() {
+    let session = get(this, 'session');
+    session.on('authenticationSucceeded', () => this.toggleConnection());
+    session.on('invalidationSucceeded', () => this.toggleConnection());
+  },
+
+  connect() {
+    if (!get(this, 'socket')) { // Only create one socket at a time
+      let token = get(this, 'session.session.authenticated.token');
+      this._super(ENV.SOCKETS_BASE_URL, {
```

Comment:
Where is this environmental variable defined?  @joshsmith

Reply:
In the config/environment.js

Reply:
Where?  I don't see it on develop branch

Reply:
The base branch is `1592-project-messages`.